sphinx_gallery/tests/test_full.py

```diff
@@ -987,7 +987,8 @@ def test_matplotlib_warning_filter(sphinx_app):
     with codecs.open(example_html, 'r', 'utf-8') as fid:
         html = fid.read()
     warning = ('Matplotlib is currently using agg, which is a'
-               ' non-GUI backend, so cannot show the figure.')
+               ' non-GUI backend, so cannot show the figure.'
+               '|(\n|.)*is non-interactive, and thus cannot be shown')
     assert warning not in html
```

Comment:
This one is not regex checked, but needs to be now that you've regex-ified the str

Reply:
... alternatively it might be easier to just keep the lines from `master` and add these two extra lines:
```
warning = "is non-interactive, and thus cannot be shown"
assert warning not in html
```

Reply:
Yes they almost have to pass regardless of actual behavior of sphinx-gallery and it's warning catching... You are currently not checking a regex match here. That would be something like:
```
assert re.match(...) is None
```
Right now it's just a `str.__in__` check:
```
assert warning not in html
```
where `warning` is a string with a bunch of regex-y characters (like `|(\n|.)`) that would never show up in the rendered HTML string when evaluated with the existing `substr in str` check.

Hence you either need to change it to `assert re.match(...) is None)`, or change it to two checks -- one which is the existing check for "old message string not in html" and an added one "new message string not in html" like: